internal/appsec/waf/waf_test.go

```diff
@@ -955,6 +1055,193 @@ func TestEncoder(t *testing.T) {
 	}
 }
 
+// This test needs a working encoder to function properly, as it first encodes the objects before decoding them
+func TestDecoder(t *testing.T) {
+	const intSize = 32 << (^uint(0) >> 63) // copied from recent versions of math.MaxInt
+	const maxInt = 1<<(intSize-1) - 1      // copied from recent versions of math.MaxInt
+	e := encoder{
+		maxDepth:        maxInt,
+		maxStringLength: maxInt,
+		maxArrayLength:  maxInt,
+		maxMapLength:    maxInt,
+	}
+	objBuilder := func(v interface{}) *wafObject {
+		var err error
+		obj := &wafObject{}
+
+		if v, ok := v.(int64); ok {
+			obj.setInt64(toCInt64(int(v)))
+			return obj
+		}
+		if v, ok := v.(uint64); ok {
+			obj.setUint64(toCUint64(uint(v)))
+			return obj
+		}
```

Comment:
Worth a comment - it took me some time to get why you do this: we're avoiding the default integer conversion of the encoder.
You can point to https://github.com/DataDog/libddwaf/issues/41 if you think it makes sense too.

Reply:
Add a comment with the link.